core/codegen/src/ident.rs

```diff
@@ -0,0 +1,99 @@
+//! A variant of Ident with more ergonomic handling of raw identifiers.
+
+use std::cmp::Ordering;
+
+use devise::syn;
+use self::syn::ext::IdentExt as _;
+use crate::proc_macro2::{Ident as Ident2, Span};
+use quote::ToTokens;
+
+#[repr(transparent)]
```

Comment:
Why have the `repr(transparent)`? Presumably we'd do this so we can cast between `&syn::Ident` and `&crate::Ident`, but we don't seem to do that.